db/pebble/transaction.go

```diff
@@ -93,3 +97,49 @@ func (t *Transaction) Impl() any {
 		return nil
 	}
 }
+
+// NewIterator : see db.Transaction.NewIterator
+func (t *Transaction) NewIterator() (db.Iterator, error) {
+	var iter *pebble.Iterator
+	if t.batch != nil {
+		iter = t.batch.NewIter(nil)
+	} else if t.snapshot != nil {
+		iter = t.snapshot.NewIter(nil)
+	} else {
+		return nil, errors.New("discarded txn")
+	}
+
+	return &pebbleIterator{iter: iter}, nil
+}
+
+// Valid : see db.Transaction.Iterator.Valid
+func (it *pebbleIterator) Valid() bool {
+	return it.iter.Valid()
+}
+
+// Key : see db.Transaction.Iterator.Key
+func (it *pebbleIterator) Key() []byte {
+	return it.iter.Key()
+}
+
+// Value : see db.Transaction.Iterator.Value
+func (it *pebbleIterator) Value() []byte {
+	return it.iter.Value()
+}
+
+// Next : see db.Transaction.Iterator.Next
+func (it *pebbleIterator) Next() bool {
+	return it.iter.Next()
+}
+
+// Seek : see db.Transaction.Iterator.Seek
+func (it *pebbleIterator) Seek(key []byte) bool {
+	it.iter.SetBounds(key, nil)
```

Comment:
Let's not restrict the view of the Iterator by setting bounds, we could need a reverse Iterator at some point.

https://pkg.go.dev/github.com/cockroachdb/pebble?utm_source=godoc#Iterator.SeekGE should do fine